src/legacy/format.ts

```diff
@@ -8,125 +8,82 @@ import {defineMessages} from 'react-intl'
-export const decodeHexAscii = (text: string) => {
+
+const decodeHexAscii = (text: string) => {
   const bytes = [...Buffer.from(text, 'hex')]
   const isAscii = bytes.every((byte) => byte > 32 && byte < 127)
   return isAscii ? String.fromCharCode(...bytes) : undefined
 }
-export const getTicker = (token: Token | DefaultAsset) => token.metadata.ticker
-export const getSymbol = (token: Token | DefaultAsset) =>
+
+const getTicker = (token: Token | DefaultAsset) => token.metadata.ticker
+
+const getSymbol = (token: Token | DefaultAsset) =>
   token.metadata.ticker
     ? utfSymbols.CURRENCIES[token.metadata.ticker]
       ? utfSymbols.CURRENCIES[token.metadata.ticker]
       : token.metadata.ticker
     : null
-export const getName = (token: Token | DefaultAsset) =>
+
+const getName = (token: Token | DefaultAsset) =>
   token.metadata.longName ||
   decodeHexAscii(token.metadata.assetName) ||
   getTokenFingerprint({
     policyId: token.metadata.policyId,
     assetNameHex: token.metadata.assetName,
   }) ||
   undefined
-// NOTE: There is a bug when starting fresh, the metadata is empty
-export const getAssetDenomination = (
-  token: Token | DefaultAsset,
-  denomination: AssetDenomination,
-): string | null | undefined => {
-  switch (denomination) {
-    case ASSET_DENOMINATION.TICKER:
-      return getTicker(token)
-
-    case ASSET_DENOMINATION.SYMBOL:
-      return getSymbol(token)
-
-    case ASSET_DENOMINATION.NAME:
-      return getName(token)
-
-    case ASSET_DENOMINATION.FINGERPRINT:
-      return getTokenFingerprint({
-        policyId: token.metadata.policyId,
-        assetNameHex: token.metadata.assetName,
-      })
-
-    default:
-      return null
-  }
-}
-export const getAssetDenominationOrId = (token: Token | DefaultAsset, denomination?: AssetDenomination): string => {
-  if (denomination !== undefined) {
-    return (
-      getAssetDenomination(token, denomination) ??
-      getTokenFingerprint({
-        policyId: token.metadata.policyId,
-        assetNameHex: token.metadata.assetName,
-      })
-    )
-  }
 
-  return (
-    getAssetDenomination(token, ASSET_DENOMINATION.TICKER) ||
-    getAssetDenomination(token, ASSET_DENOMINATION.NAME) ||
-    getTokenFingerprint({
-      policyId: token.metadata.policyId,
-      assetNameHex: token.metadata.assetName,
-    })
-  )
-}
-export const getAssetDenominationOrUnknown = (
-  token: Token | DefaultAsset,
-  denomination: AssetDenomination,
-  intl: IntlShape,
-): string => getAssetDenomination(token, denomination) ?? intl.formatMessage(messages.unknownAssetName)
 export const normalizeTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): BigNumber => {
   const normalizationFactor = Math.pow(10, token.metadata.numberOfDecimals)
   return amount.dividedBy(normalizationFactor).decimalPlaces(token.metadata.numberOfDecimals)
 }
+
 export const formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string => {
```

Comment:
Maybe overload it, to work closer with Quantities/Entries/Amounts?

formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string
formatTokenAmount = (amount: string, token: Token | DefaultAsset): string
```suggestion
export const formatTokenAmount = (amount: BigNumber, token: Token | DefaultAsset): string => {
```